pkg/cmd/cli/completion/completion.go

```diff
@@ -0,0 +1,207 @@
+/*
+Copyright 2017 the Heptio Ark contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package completion
+
+import (
+	"fmt"
+	"os"
+
+	"bytes"
+	"github.com/spf13/cobra"
+	"io"
+)
+
+func NewCommand() *cobra.Command {
+	c := &cobra.Command{
+		Use:   "completion SHELL",
+		Short: "Output shell completion code for the specified shell (bash or zsh)",
+		Long: `Generate shell completion code.
+
+Auto completion supports both bash and zsh. Output is to STDOUT.
+
+Load the ark completion code for bash into the current shell -
+source <(ark completion bash)
+
+Load the ark completion code for zsh into the current shell -
+source <(ark completion zsh)
+`,
+		Args:      cobra.ExactArgs(1),
+		ValidArgs: []string{"bash", "zsh"},
+		Run: func(cmd *cobra.Command, args []string) {
+			shell := args[0]
+			switch shell {
+			case "bash":
+				cmd.Root().GenBashCompletion(os.Stdout)
+			case "zsh":
+				genZshCompletion(os.Stdout, cmd.Root())
+			default:
+				fmt.Printf("Invalid shell specified, specify bash or zsh\n")
+				os.Exit(1)
+			}
+		},
+	}
+
+	return c
+}
+
+func genZshCompletion(out io.Writer, ark *cobra.Command) {
+	// The following code has been copied from
```

Comment:
We probably need to move this to `third_party` and include the license header from the original file. Let me double check and get back to you later.

Reply:
sure

Reply:
ping @ncdc

Reply:
If you can find a way to put this in `third_party` under `kubernetes/pkg/kubectl/cmd/completion.go` along with the original license and a line saying that modifications are © Heptio Ark Contributors, that'd probably be best.

Reply:
fixed

Reply:
I've actually included both the licences in the new file in third_party, PTAL